lib/core/enums/PublicKeyUsage.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Sidetree public key usage.
+ */
+enum PublicKeyUsage {
```

Comment:
You might consider aligning these with https://w3c-ccg.github.io/ld-proofs/#proof-purpose

or https://www.w3.org/TR/WebCryptoAPI/#cryptokey-interface-types

or maybe not using the term general term "usage".... for example, these are really sidetreeProtocolKeyUsage... these keys might have uses according to existing standards linked above, beyond just what sidetree is doing.